prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -320,35 +321,49 @@ void rxr_pkt_req_data_from_tx(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry
 		return;
 	}
 
+	base_hdr = rxr_get_base_hdr(pkt_entry->pkt);
+	/*
+	 * For eager msg, we always copy data to packet entry.
+	 * This way, we can write tx completion as soon as the packet
+	 * has been sent. Because data has been copied to our bounce
+	 * buffer, application can safely re-use its send buffer.
+	 */
+	if (base_hdr->type == RXR_EAGER_MSGRTM_PKT ||
+	    base_hdr->type == RXR_EAGER_TAGRTM_PKT)
+		goto copy;
+
 	rxr_locate_iov_pos(tx_entry->iov, tx_entry->iov_count, data_offset,
 			   &tx_iov_index, &tx_iov_offset);
-	desc = tx_entry->desc[0];
 	assert(tx_iov_index < tx_entry->iov_count);
 	assert(tx_iov_offset < tx_entry->iov[tx_iov_index].iov_len);
 
 	/*
-	 * Copy can be avoid if:
-	 * 1. user provided memory descriptor, or lower provider does not need memory descriptor
-	 * 2. data to be send is in 1 iov, because device only support 2 iov, and we use
-	 *    1st iov for header.
+	 * We have to copy if:
```

Comment:
I like the old phrasing better.  Highlight the paths when we can avoid the copy.  Copies are bad.  Except when they're not, but they're still bad :).

Reply:
Changed back to old phrasing in new revision.